js/alerts.js

```diff
@@ -53,41 +100,106 @@ function loadSelectors() {
 
     });
 
-    $('#button').click(function () {
-        var rev = $('#rev').val();
-        var test = $('#test').val();
-        var platform = $('#platform').val();
-        var showall = 0;
-        var testIndex = $("select[id='test'] option:selected").index();
-        var platIndex = $("select[id='platform'] option:selected").index();
-        if ($('#checkbox').is(":checked")) {
-            console.log("checked");
-            showall = 1;
-        }
-        document.cookie = "platform = " + platform;
-        document.cookie = "test = " + test;
+    $('#button').bind("click", filterOnClick);
+}
 
-        var href = "alerts.html";
-        var flag = '?';
-        if (rev && rev != '') {
-            href += flag + "rev=" + rev;
-            flag = '&';
+//used for lexical sorting
+function compare(a, b) {
+    a = a.toString().toLowerCase();
+    b = b.toString().toLowerCase();
+    if (a < b)
+        return -1;
+    if (a > b)
+        return 1;
+    return 0;
+}
+
+function resetOptions(data,id) {
+    for (var i in data) {
+        var newoption = document.createElement("option");
+        newoption.id = id;
+        var value = data[i];
+        $("#"+id).append("<option value=\"" + value + "\">" + value + "</option>");
+    }
+}
+
+function updateSelectors(changedElementId, callback) {
+    
+    //to prevent selecting the button while the options are being updated [ to avoid irregular indexes of options ]
+    $('#button').unbind("click", filterOnClick).attr('disabled','disabled');
+    
+    rev = $('#rev').val() || "";
+    test = $('#test').val() || "";
+    platform = $('#platform').val() || "";
+    
+    if ( callback && typeof(callback)==="function") {
+        //callback function is used only when filter button click event happens
+        //we want to re-order the list of options according to the revision selection only
+        // so that the indexes i.e. testindex and platindex are uniform when the result is loaded using URL
+        get_params = {
+            'name': ['keyrevision'],
+            'value': [rev]
+        };
+    }
+    else {
+       get_params = { 
+            'name': [ 'keyrevision', 'test', 'platform' ],
+            'value': [ rev, test, platform ]
+        }; 
+    }
+    
+    var elements = { 'rev': '#rev', 'test': '#test', 'platform': '#platform' };
+    var attribute = elements[changedElementId];
+
+    //to prevent from choosing value while options are being updated
+    for (var key in elements) {
+        if (key != changedElementId)
+            $(elements[key]).attr('disabled','disabled');
+    }
+    
+    $.getJSON(root_url+"/getvalues", $.param(get_params, true),function(data) {
+        
+        var tests = data['test'].sort(compare);
+        var revs = data['rev'];
+        var platforms = data['platform'].sort(compare);
+
+        if (changedElementId != "rev") {
```

Comment:
what about a for loop here as well
for (var key in elements) {
  if (changedElementId != key) {
    $("#" + key).children().remove().end().append('<option value="">Select ' + key.toUpperCase() + '</option>');

you can see how this cleans it up quite well.  key will be 'rev' instead of 'revision', so maybe expand the little hash of elements.  You could replace it and use the '#' + key, trick like I did above.